### ISSUE TITLE
bug(datepicker):  Missing to announce "to" in reading year range. Ex. "2016-2039"

### ISSUE THREAD

**Issue by zarend** [angular/components#23467]
#### Reproduction

Steps to reproduce:
1. Open the [Datepicker Examples](https://material.angular.io/components/datepicker/examples)
2. Go to the "Basic date range picker" demo
3. Open the calendar dialog
4. Focus and activate on the "Choose month and year"
5. Observe the audio feedback
 
 
#### Expected Behavior
It should read out "to" between the years by the screen reader

#### Actual Behavior
It is reading "2016 2039" without mentioning either "-" or "to"

#### Environment
OS| Browser | Screen reader
ChromeOS | Google Chrome 91.0.4472.167 | ChromeVox
macOS Catalina | Google Chrome 92.0.4515.131 | VoiceOver

##### Additional Notes
This is copied from an [internal bug report](http://b/195600354) found during a11y testing.
